File: template-repo/templates/uniapp-springboot-api/backend/src/main/java/com/smartark/template/mobile/ai/PromptBuilder.java
package com.smartark.template.mobile.ai;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Simple prompt template builder with {{variable}} substitution.
 *
 * Usage:
 *   String prompt = PromptBuilder.of("Analyze {{document}} for {{purpose}}")
 *       .set("document", resumeText)
 *       .set("purpose", "job matching")
 *       .build();
 */
public class PromptBuilder {

    private String template;
    private final Map<String, String> variables = new LinkedHashMap<>();

    public static PromptBuilder of(String template) {
        PromptBuilder builder = new PromptBuilder();
        builder.template = template;
        return builder;
    }

    public PromptBuilder set(String key, String value) {
        variables.put(key, value != null ? value : "");
        return this;
    }

    public String build() {
        String result = template;
        for (Map.Entry<String, String> entry : variables.entrySet()) {
            result = result.replace("{{" + entry.getKey() + "}}", entry.getValue());
        }
        return result;
    }
}

File: template-repo/templates/nextjs-mysql/lib/ai/PromptBuilder.ts
import 'server-only'

export class PromptBuilder {
  private readonly template: string
  private readonly variables = new Map<string, string>()

  constructor(template: string) {
    this.template = template
  }

  set(key: string, value: string | null | undefined) {
    this.variables.set(key, value ?? '')
    return this
  }

  build() {
    let result = this.template
    for (const [key, value] of this.variables.entries()) {
      result = result.replaceAll(`{{${key}}}`, value)
    }
    return result
  }
}
